thrift/src/main/java/io/grpc/thrift/MessageFactory.java

```diff
@@ -31,11 +31,14 @@
 
 package io.grpc.thrift;
 
+import io.grpc.ExperimentalApi;
+
 import org.apache.thrift.TBase;
 
 /** 
  *  Produce new Message Instances. Used by Marshaller to deserialize incoming messages. 
  */
+@ExperimentalApi("https://github.com/grpc/grpc-java/issues/2170")
```

Comment:
Why? I'm missing something. Annotations generally go right next to the declaration, not separated by a comment.